python/ray/air/tests/test_keras_callback.py

```diff
@@ -1,21 +1,28 @@
 import os
+import sys
 from typing import Dict, Tuple
 from unittest.mock import patch
 
 import numpy as np
 import pytest
-import tensorflow as tf
 
 import ray
 from ray import train
-from ray.air.integrations.keras import ReportCheckpointCallback
 from ray.train import ScalingConfig
 from ray.train.constants import TRAIN_DATASET_KEY
-from ray.train.tensorflow import (
-    TensorflowCheckpoint,
-    TensorflowPredictor,
-    TensorflowTrainer,
-)
+
+if sys.version_info >= (3, 12):
+    # Tensorflow is not compatible with Python 3.12
```

Comment:
tensorflow does support py312. I think there are just issues with `keras <> tf <> ray`. Maybe we should skip if  `tf version > 2.16` instead?

Reply:
nice, let me update the comments; i don't even install tensorflow for python 3.12 on CI (save me the complexity on resolving conflicts)